app/controllers/thread_controller.php

```diff
@@ -0,0 +1,97 @@
+<?php 
+class ThreadController extends AppController
+{
+    public function create()
+    {
+        $thread = new Thread();
+        $comment = new Comment();
+        
+        $page = Param::get('page_next', 'create');
+
+        switch ($page) {
+            case 'create':
+                break;
+            case 'create_end':
+                $thread->title = Param::get('title');
+                $comment->username = Param::get('username');
+                $comment->body = Param::get('body');
+
+                try {
+                    $thread->create($comment);
+                } catch (ValidationException $e) {
+                    $page = 'create';
+                }
+                break;
+            default:
+                throw new NotFoundException("{page} is not found");
+                break;
+        }
+
+        $this->set(get_defined_vars());
+        $this->render($page);
+    }
+
+    public function index()
+    {
+        $per_page = 5;
+        $page = Param::get('page', 1);
+
+        $pagination = new SimplePagination($page, $per_page);
+
+        $threads = Thread::getAll($pagination->start_index -1, $pagination->count + 1);
+        $pagination->checkLastPage($threads);
+       
+        $total = Thread::countAll();
```

Comment:
Instead of creating a new method for counting all threads, you can just use the count method
```php
$total = count($threads);
```